RecoMET/METProducers/src/HcalNoiseInfoProducer.cc

```diff
@@ -730,40 +730,42 @@ HcalNoiseInfoProducer::filldigis(edm::Event& iEvent, const edm::EventSetup& iSet
        // (if this was not the case, then we just end up using 
        // all data from all fibers )
        for( unsigned fidx = 0; fidx < (nFibers - 1); ++fidx ) {
+
          // start with the last TS and loop backwards
          // on each iteration check if all capId and ADCs from this
          // TS forward match the beginning entries of 
          // the next fiber
-         int last_ts = lasmon_capids[fidx].size()-1;  // last TS
+         unsigned last_ts = lasmon_capids[fidx].size()-1;  // last TS
          int start_ts = last_ts; // start_ts will be decrimented on each loop
          // in the case that our stringent check below doesn't work 
          // store the latest capID that has a match
          int latest_cap_match = -1;
          // reverse loop over TSs
-         while( start_ts >= 0 ) {
-           int ncheck = 0; // count the number of TS that were checked
-           int nmatch_cap = 0; // count the number of TS where capID matched
-           int nmatch_adc = 0; // count the number of TS where ADC matched
-
-           for( unsigned cidx = (unsigned)start_ts, nidx=0; 
-                   cidx <= (unsigned)last_ts; cidx++, nidx++ ) {
-             ncheck++;
+         for( unsigned ncheck = 1; ncheck <= lasmon_capids[fidx].size() ; ncheck++ ) {
+           unsigned nmatch_cap = 0; // count the number of TS where capID matched
+           unsigned nmatch_adc = 0; // count the number of TS where ADC matched
+
+           // loop over the channel TS, this is for the later fiber in time
+           for( unsigned lidx = 0; lidx < ncheck; lidx++) {
+             // we are looping over the TS of the later fiber in time
+             // the TS of the earlier fiber starts from the end
+             unsigned eidx = last_ts-ncheck+lidx+1;
```

Comment:
```
            unsigned eidx = last_ts-ncheck+lidx;
```
In principle this is risky, but ncheck is bound to last_ts at L744, therefore it works